s3-native-client/src/test/java/com/amazonaws/s3/S3NativeClientTest.java

```diff
@@ -160,6 +168,115 @@ public void onException(final CrtRuntimeException e) {
+
+        try (final EventLoopGroup elGroup = new EventLoopGroup(9);
+                final HostResolver resolver = new HostResolver(elGroup, 128);
+                final ClientBootstrap clientBootstrap = new ClientBootstrap(elGroup, resolver);
+                final CredentialsProvider provider = getTestCredentialsProvider()) {
+
+            Exception exceptionResult = null;
+
+            final S3NativeClient nativeClient = new S3NativeClient(REGION, clientBootstrap, provider, 64_000_000l,
+                    100.);
+
+            final CancelTestData<GetObjectOutput> testData = new CancelTestData<GetObjectOutput>();
+            testData.ExpectedPartCount = 1;
+
+            try {
+                testData.ResultFuture =
+                         nativeClient.getObject(GetObjectRequest.builder().bucket(BUCKET).key(GET_OBJECT_KEY).build(),
+                                                new ResponseDataConsumer<GetObjectOutput>() {
+
+                            @Override
+                            public void onResponse(GetObjectOutput response) {
+                                assertNotNull(response);
+                            }
+
+                            @Override
+                            public void onResponseData(ByteBuffer bodyBytesIn) {
+                                ++testData.PartCount;
+
+                                if (testData.PartCount == testData.ExpectedPartCount) {
+                                    testData.ResultFuture.cancel(true);
+                                }
+                            }
+
+                            @Override
+                            public void onFinished() {
+                            }
+
+                            @Override
+                            public void onException(final CrtRuntimeException e) {
+                            }
+                        });
+                testData.ResultFuture.join();
+            } catch (Exception e) {
+                exceptionResult = e;
+            }
+
+            Assume.assumeTrue(exceptionResult != null);
+            Assume.assumeTrue(exceptionResult instanceof CancellationException);
+            Assume.assumeTrue(testData.PartCount == testData.ExpectedPartCount);
+        }
+    }
+
+    @Test
+    public void testPutObjectCancel() {
+        Assume.assumeTrue(System.getProperty("NETWORK_TESTS_DISABLED") == null);
+
+        try (final EventLoopGroup elGroup = new EventLoopGroup(9);
+                final HostResolver resolver = new HostResolver(elGroup, 128);
+                final ClientBootstrap clientBootstrap = new ClientBootstrap(elGroup, resolver);
+                final CredentialsProvider provider = getTestCredentialsProvider()) {
+
+            Exception exceptionResult = null;
+
+            final S3NativeClient nativeClient = new S3NativeClient(REGION, clientBootstrap, provider, 64_000_000l,
+                    100.);
+            final long contentLength = 1 * 1024l * 1024l * 1024l;
```

Comment:
hmm

Reply:
Changed this to be in relation to part sizes.